### DISCUSSION TITLE
Select自定义弹窗里的Dropdown 点击时,如何阻止select收起?

### DISCUSSION THREAD

**Discussion by vaynevayne** [ant-design/ant-design#42678] [Q&A]
https://stackblitz.com/edit/react-g1rqy7?file=demo.tsx,tsconfig.json

受控的open,  点击时强制设置setOpen(false) 也不行

**Comment by BoyYangzai** [comment]
@vaynevayne 
https://stackblitz.com/edit/react-g1rqy7-tgksxw?file=demo.tsx,index.css
这样呢 如果想要加上点击外面关闭 可以再加上 `useClickOutsides`

```ts
export function useClickOutsides(refCurrents: HTMLDivElement[], callback: () => void) {
  useEffect(() => {
    function handler(event: MouseEvent) {
      if (refCurrents.some(item=>item!==null) &&!refCurrents.some(item => item?.contains(event.target! as Node))) {
        callback();
      }
    }
    window.addEventListener('click', handler);

    return () => window.removeEventListener('click', handler)
  }, [callback, refCurrents]);
}
```

`refCurrents` 传入你点击不想关闭的 dom

**Comment by afc163** [comment]
https://ant.design/components/select-cn#%E5%8F%8D%E8%BF%87%E6%9D%A5%E5%B8%8C%E6%9C%9B%E7%82%B9%E5%87%BB-dropdownrender-%E9%87%8C%E5%85%83%E7%B4%A0%E4%B8%8D%E6%B6%88%E5%A4%B1%E8%AF%A5%E6%80%8E%E4%B9%88%E5%8A%9E

**Reply by vaynevayne** [reply]
@afc163  dropdownRender 里面如果有  PopConfirm ,这个方式就不起作用, 点击PopConfirm 还是会关闭

**Comment by vaynevayne** [comment]
多谢两位
